### PR TITLE
Add types to HTTP::StaticFileHandler

### PR DESCRIPTION
**Pull request by jkthorne**
This adds types to StaticFuleHandler initialize and move the assignment to the arg assignemnt

### CODE REVIEW

**Comment by straight-shoota** [comment]
Hm, this is technically a breaking change because things `HTTP::StaticFileHandler.new("", fallthrough: nil)` did work before this change. I don't think there would be any reasonable justification for that, but it's possible 🤷 

So maybe we should add an unrestricted overload just to be safe?
```cr
# :ditto:
@[Deprecated]
def self.new(public_dir : String, fallthrough = true, directory_listing = true)
  new(public_dir, fallthrough: fallthrough, listing: listing)
end
```

File: src/http/server/handlers/static_file_handler.cr [hunk 1]
```diff
@@ -31,10 +31,14 @@ class HTTP::StaticFileHandler
...
+  def initialize(public_dir : String, @fallthrough : Bool = true, @directory_listing : Bool = true)
     @public_dir = Path.new(public_dir).expand
-    @fallthrough = !!fallthrough
-    @directory_listing = !!directory_listing
+  end
+
+  # :ditto:
+  @[Deprecated]
+  def self.new(public_dir : String, fallthrough = true, directory_listing = true)
+    new(public_dir, fallthrough: fallthrough, listing: listing)
```
Comment (Sija) [inline src/http/server/handlers/static_file_handler.cr:41]:
```suggestion
    new(public_dir, fallthrough: !!fallthrough, listing: !!listing)
```

**Review by straight-shoota — APPROVED** [review]

**Comment by Blacksmoke16** [comment]
Why add the breaking change label again? It should no longer be due to the new overload.

**Comment by straight-shoota** [comment]
It adds a depreciation and deprecations are marked as breaking changes.

**Comment by HertzDevil** [comment]
Don't we now have a separate deprecation tag

**Comment by straight-shoota** [comment]
Oops right. Guess I was out on vacation for too long 😏

**Review by beta-ziliani — APPROVED** [review]

**Pull request merged**